src/Admin/entrypoint.sh

```diff
@@ -5,52 +5,27 @@
 GROUPNAME="bitwarden"
 USERNAME="bitwarden"
 
-CURRENTGID=`getent group $GROUPNAME | cut -d: -f3`
-LGID=${LOCAL_GID:-999}
+LUID=${LOCAL_UID:-0}
+LGID=${LOCAL_GID:-0}
 
-NOUSER=`id -u $USERNAME > /dev/null 2>&1; echo $?`
-LUID=${LOCAL_UID:-999}
+# Step down from host root to well-known nobody/nogroup user
 
-# Step down from host root
-
-if [ $LGID == 0 ]
+if [ $LUID -eq 0 ]
 then
-    LGID=999
+    LUID=65534
 fi
-
-if [ $LUID == 0 ]
+if [ $LGID -eq 0 ]
 then
-    LUID=999
+    LGID=65534
 fi
 
-# Create group
+# Create user and group
 
-if [ $CURRENTGID ]
-then
-    if [ "$CURRENTGID" != "$LGID" ]
-    then
-        groupmod -g $LGID $GROUPNAME
-    fi
-else
-    groupadd -g $LGID $GROUPNAME
-fi
-
-# Create user and assign group
-
-if [ $NOUSER == 0 ] && [ `id -u $USERNAME` != $LUID ]
-then
-    usermod -u $LUID $USERNAME
-elif [ $NOUSER == 1 ]
-then
-    useradd -r -u $LUID -g $GROUPNAME $USERNAME
-fi
-
-# Make home directory for user
-
-if [ ! -d "/home/$USERNAME" ]
-then
-    mkhomedir_helper $USERNAME
-fi
+groupadd -o -g $LGID $GROUPNAME >/dev/null 2>&1 ||
+groupmod -o -g $LGID $GROUPNAME >/dev/null 2>&1
+useradd -o -u $LUID -g $GROUPNAME -s /bin/false $USERNAME >/dev/null 2>&1 ||
```

Comment:
Why is the `-s /bin/false` necessary? Not sure I understand what that is doing here, even after reading the man page on `-s`.

Reply:
When you add a user to a system, you give it a login shell, which will be used when this user will connect to the system.
It could be `bash`, `sh`, `ksh`...
Service users are not intended to login, thus we usually give us a non-shell, such as `/bin/false` or `/usr/sbin/nologin`.
This is the last parameter of the `/etc/passwd` file.
I then added `-s /bin/false` to the command, to be consistent, without there was no shell set for the user.